Framework/DataObjects/src/Peak.cpp

```diff
@@ -222,24 +170,8 @@ Peak::Peak(const Peak &other)
  * @return
  */
 Peak::Peak(const Geometry::IPeak &ipeak)
-    : IPeak(ipeak), m_detectorID(ipeak.getDetectorID()), m_H(ipeak.getH()),
-      m_K(ipeak.getK()), m_L(ipeak.getL()), m_intensity(ipeak.getIntensity()),
-      m_sigmaIntensity(ipeak.getSigmaIntensity()),
-      m_binCount(ipeak.getBinCount()),
-      m_initialEnergy(ipeak.getInitialEnergy()),
-      m_finalEnergy(ipeak.getFinalEnergy()),
-      m_absorptionWeightedPathLength(ipeak.getAbsorptionWeightedPathLength()),
-      m_GoniometerMatrix(ipeak.getGoniometerMatrix()),
-      m_InverseGoniometerMatrix(ipeak.getGoniometerMatrix()),
-      m_runNumber(ipeak.getRunNumber()),
-      m_monitorCount(ipeak.getMonitorCount()), m_row(ipeak.getRow()),
-      m_col(ipeak.getCol()), m_peakNumber(ipeak.getPeakNumber()),
-      m_intHKL(ipeak.getIntHKL()), m_intMNP(ipeak.getIntMNP()),
-      m_peakShape(std::make_shared<NoShape>()) {
+    : BasePeak(ipeak), m_detectorID(ipeak.getDetectorID()) {
```

Comment:
Would it be useful to add a constructor that takes a `LeanPeak` and some instrument info?

Reply:
I already have a task for this, SCD196